spread/runner.go

```diff
@@ -594,27 +595,32 @@ func (r *Runner) worker(backend *Backend, system *System, order []int) {
 		}
 
 		debug := job.Debug()
-		if r.options.Restore {
-			// Do not prepare or execute.
-		} else if !r.options.Restore && !r.run(client, job, preparing, job, job.Prepare(), debug, &abend) {
-			r.add(&stats.TaskPrepareError, job)
-			r.add(&stats.TaskAbort, job)
-			debug = ""
-		} else if !r.options.Restore && r.run(client, job, executing, job, job.Task.Execute, debug, &abend) {
-			r.add(&stats.TaskDone, job)
-		} else if !r.options.Restore {
-			r.add(&stats.TaskError, job)
-			debug = ""
-		}
-		if !abend && !r.options.Restore {
-			if err := r.fetchResidue(client, job); err != nil {
-				printf("Cannot fetch residue of %s: %v", job, err)
-				r.tomb.Killf("cannot fetch residue of %s: %v", job, err)
+		for repeat := r.options.Repeat; repeat >= 0; repeat-- {
+			if r.options.Restore {
+				// Do not prepare or execute.
```

Comment:
```
// Do not prepare or execute, and don't repeat.
repeat = -1
```
We might conflict the options at some point, but for now it at least doesn't make sense to repeat the restore phase alone.